src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,87 @@
+import { useState } from 'react';
+import classnames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { useTodo } from '../../context/TodoContext';
+
+type Props = {
+  todo: Todo;
+  onDelete: (value: Todo) => void;
+  onEditedId: (value: number | null) => void;
+  isDeleteActive: boolean;
+  onDeleteActive?: (value: boolean) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo, onDelete, onEditedId, isDeleteActive,
+}) => {
+  const { setTodos } = useTodo();
+  const { id, title, completed } = todo;
+
+  const [deletedTodoId, setDeletedTodoId] = useState<number | null>(null);
+
+  const handleComplete = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setTodos(prev => prev.map(currentTodo => {
+      if (currentTodo.id === todo.id) {
+        return { ...currentTodo, completed: event.target.checked };
+      }
+
+      return currentTodo;
+    }));
```

Comment:
You don't update the todo on the server, when I reload page status doesn't change